services/thermal_mgr/thermal_mgr.c

```diff
@@ -24,48 +24,107 @@ static uint8_t thermalMgrQueueStorageArea[THERMAL_MGR_QUEUE_LENGTH * THERMAL_MGR
 
 static void thermalMgr(void *pvParameters);
 
-void initThermalSystemManager(lm75bd_config_t *config) {
+void initThermalSystemManager(lm75bd_config_t *config)
+{
   memset(&thermalMgrTaskBuffer, 0, sizeof(thermalMgrTaskBuffer));
   memset(thermalMgrTaskStack, 0, sizeof(thermalMgrTaskStack));
-  
+
   thermalMgrTaskHandle = xTaskCreateStatic(
-    thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
-    config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
+      thermalMgr, "thermalMgr", THERMAL_MGR_STACK_SIZE,
+      config, 1, thermalMgrTaskStack, &thermalMgrTaskBuffer);
 
   memset(&thermalMgrQueueBuffer, 0, sizeof(thermalMgrQueueBuffer));
   memset(thermalMgrQueueStorageArea, 0, sizeof(thermalMgrQueueStorageArea));
 
   thermalMgrQueueHandle = xQueueCreateStatic(
-    THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
-    thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
-
+      THERMAL_MGR_QUEUE_LENGTH, THERMAL_MGR_QUEUE_ITEM_SIZE,
+      thermalMgrQueueStorageArea, &thermalMgrQueueBuffer);
 }
 
-error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
+error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event)
+{
+  /* Send an event to the thermal manager queue */
   /* Send an event to the thermal manager queue */
 
-  return ERR_CODE_SUCCESS;
+  // check to make sure the queue has been created already
+  if (event == NULL)
+    return ERR_CODE_INVALID_QUEUE_MSG;
+  if (thermalMgrQueueHandle == NULL)
+  {
+    return ERR_CODE_INVALID_ARG; // Return an error code if the queue hasn't been created
+  }
+
+  if (xQueueSend(thermalMgrQueueHandle, (void *)event, (TickType_t)0) == pdTRUE)
+  {
+    return ERR_CODE_SUCCESS;
+  }
+  else
+    return ERR_CODE_QUEUE_FULL; // If it doesn't send, then queue is full
 }
 
-void osHandlerLM75BD(void) {
+void osHandlerLM75BD(void)
+{
   /* Implement this function */
+  thermal_mgr_event_t interruptEvent;
+  interruptEvent.type = THERMAL_MGR_EVENT_INTERRUPT;
+  thermalMgrSendEvent(&interruptEvent);
 }
 
-static void thermalMgr(void *pvParameters) {
+static void thermalMgr(void *pvParameters)
+{
   /* Implement this task */
-  while (1) {
-    
+
+  while (1)
+  {
+    thermal_mgr_event_t receivedData;
+    float currTemp = 0;
+    error_code_t checker;
+    if (xQueueReceive(thermalMgrQueueHandle, &receivedData, portMAX_DELAY) == pdTRUE)
+    {
+      switch (receivedData.type)
+      {
+      case THERMAL_MGR_EVENT_MEASURE_TEMP_CMD:
+        checker = readTempLM75BD(LM75BD_OBC_I2C_ADDR, &currTemp);
+        if (checker != ERR_CODE_SUCCESS)
+        {
+          return checker;
```

Comment:
this is a void function so you should not be return a value, and since this is a task function you don't want to return and kill the task just because on one error so just `continue` instead